runtime.js

```diff
@@ -14,11 +16,16 @@ function checkProcessArgv (moduleName) {
 
 let preloadModules
 function checkPreloadModules (moduleName) {
+  const modulePath = path.join(process.cwd(), 'node_modules/', moduleName)
+
   /* c8 ignore start */
-  // nullish needed for non Node.js runtime
+  // coverage - nullish needed for non Node.js runtime
   preloadModules ??= (process._preload_modules ?? [])
+
+  // coverage - TS specific
+  return preloadModules.includes(moduleName) ||
+    Object.keys(require.cache).some(k => k.startsWith(modulePath) && preloadModules.push(modulePath))
```

Comment:
It's not a good idea to mutate `preloadModules` in `Array.some()` and relying the truthy return value of `Array.push()`.
By the way, pushing `modulePath` instead of `moduleName` here is kinda incorrect.

Reply:
It's weird because I overthought this to avoid a couple of lines and didn't found an obvious issue.

If `preloadModules.includes(moduleName)` evaluate to true, LogicalOr right expression is not evaluated.
`some` stop iterating if evaluate truthy, LogicalAnd right is always evaluated and `push` always return truthy value when adding an element.

I refactored because I think you're right, the code is misleading and can cause problems.

Reply:
>  By the way, pushing modulePath instead of moduleName here is kinda incorrect.

Fixed, thanks.